project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/AppSettings.kt

```diff
@@ -0,0 +1,46 @@
+package org.plan.research.minimization.plugin.settings
+
+import org.plan.research.minimization.plugin.model.FileLevelStage
+import org.plan.research.minimization.plugin.model.MinimizationStage
+import org.plan.research.minimization.plugin.model.state.*
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import org.jetbrains.annotations.NonNls
+
+@State(
+    name = "org.intellij.sdk.settings.AppSettings",
+    storages = [Storage("ProjectMinimizationSettings.xml")],
+)
+class AppSettings : PersistentStateComponent<AppSettings.State> {
```

Comment:
Yes, but I left this for [Handle setting change](https://youtrack.jetbrains.com/issue/JBRes-1997/Handle-Settings-Change) because it seems like fixing this duplication and connect MinimizationPluginSettings and AppSettings into one class is more related to this task.

Reply:
Ok, that works